pkgs/by-name/ko/koboldcpp/package.nix

```
@@ -0,0 +1,98 @@
+{ lib
+, fetchFromGitHub
+, stdenv
+, autoPatchelfHook
+, makeWrapper
+, python311Packages
+, tk
+
+, config
+, cudaPackages ? { }
+
+, openblasSupport ? true
+, openblas
+
+, cublasSupport ? config.cudaSupport
+
+, clblastSupport ? true
+, clblast
+, ocl-icd
+
+, vulkanSupport ? true
+, vulkan-loader
+
+, metalSupport ? stdenv.isDarwin && stdenv.isAarch64 && !openblasSupport
+}:
+
+let
+  inherit (lib) optionals;
+
+  pythonInputs = with python311Packages; [
+    tkinter
+    customtkinter
+    packaging
+    tk
+  ];
+
+  cudaBuildInputs = with cudaPackages; [
+      libcublas
+  ];
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "koboldcpp";
+  version = "1.65";
+
+  src = fetchFromGitHub {
+    owner = "LostRuins";
+    repo = "koboldcpp";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-GYvU+0bhskCgSXAoeSy/2wWIF5FrilLVdXitFvuUAdQ=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    makeWrapper
+    python311Packages.wrapPython
+  ];
+
+  buildInputs = pythonInputs ++ optionals openblasSupport [ openblas ]
+  ++ optionals cublasSupport [ cudaPackages.libcublas ]
+  ++ optionals clblastSupport [ clblast ocl-icd ]
+  ++ optionals vulkanSupport [ vulkan-loader ];
+
+  pythonPath = pythonInputs;
+
+  makeFlags = optionals openblasSupport [ "LLAMA_OPENBLAS=1" ]
+  ++ optionals cublasSupport [ "LLAMA_CUBLAS=1" ]
+  ++ optionals clblastSupport [ "LLAMA_CLBLAST=1" ]
+  ++ optionals vulkanSupport [ "LLAMA_VULKAN=1" ]
+  ++ optionals metalSupport [ "LLAMA_METAL=1" ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/run-env
+
+    install -Dm755 koboldcpp.py $out/run-env/koboldcpp-unwrapped
+    cp *.so $out/run-env
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    wrapPythonProgramsIn "$out/run-env" "$pythonPath"
+    makeWrapper $out/run-env/koboldcpp-unwrapped $out/bin/koboldcpp \
+    --prefix PATH ${lib.makeBinPath [
+      tk
+    ]}
+  '';
+
+  meta = with lib; {
+    description = "A way to run various GGML and GGUF models";
+    license = licenses.agpl3Only;
+    mainProgram = "koboldcpp";
+    platforms = platforms.unix;
+    maintainers = with maintainers; [ maxstrid ];
```

```suggestion
    maintainers = with maintainers; [ maxstrid donteatoreo ];
```

I would like to be a maintainer for `koboldcpp` as well